src/build/app/mod.rs

```diff
@@ -628,12 +629,13 @@ impl<'b> App<'b> {
     /// # ;
     /// ```
     /// [argument]: ./struct.Arg.html
-    pub fn arg<A: Into<Arg<'b>>>(mut self, a: A) -> Self {
-        let help_heading: Option<&'b str> = if let Some(option_str) = self.help_headings.last() {
-            *option_str
-        } else {
-            None
-        };
+    pub fn arg<A: Into<Arg>>(mut self, a: A) -> Self {
+        let help_heading: Option<Cow<'static, str>> =
```

Comment:
Would it be good to tag `as_deref` to be used for rust 1.40.0 later on?

Reply:
I don't understand what `as_deref` has to do with the bit of code you highlighted. Could you please elaborate?

Reply:
Something like
```rust
         let help_heading: Option<Cow<'static, str>> = self.help_headings.last().map(Deref::deref);
```

Reply:
True, thanks!

Reply:
True, thanks!